proposed/2022/Package-Source-Mapping-VS-Options.md

```diff
@@ -0,0 +1,145 @@
+
+## Motivation 
+
+Adding support for package source mapping in the VS options dialog will allow user to pin sources and add/remove package source mappings with more ease.
+
+## Explanation
+
+### Functional Explanation
+
+#### Package Source Mappings Page 
+
+When the user opens the package source mapping page in the options dialog, they will see a list of all the package source mappings they have already configured and the filepath to the solution level config file. There is a hierarchy of NuGet configs levels: machine, user, and solution. Package source mappings will be read from all of the levels, but only written to the solution level config. This page will have three buttons: _Add_ , _Remove_, and _Clear_.
+
+* Add: Brings user to a pop-up where they can add package source mappings to their project.
+
+* Remove: Lets user remove selected package source mapping from project.
+
+* Clear: Lets user clear all package source mappings from current project.
+
+The popup when the user clicks the add button will have a textbox where a user can type in a package ID. All sources that are defined in package sources page of VS options dialog will appear in a `ComboBox` below. The user checks a `CheckBox` for the sources they would like to use for that package ID. 
+
+The add window has two buttons: _Add_ and _Close_.
+
+* Add: Add button adds package source mapping to project only using selected sources, and the pop-up will disappear. 
+
+* Close: Will close pop-up without adding any package source mappings.
+
+Below are the mockups for the package source mappings page. The top left is the package source mappings page, and the bottom left is the popup that appears when the user clicks the add button.
+
+![Options 1](../../meta/resources/PackageSourceMapping/VSOptions.png)
+
+#### Package Sources Page
+
+I will add a feature that allows a user to pin default sources for all package source mappings that are not specified on the package source mapping page. There will be a pin button, that when clicked, creates a default mapping for all package IDs to the pinned source. A pin icon will appear next to all pinned sources.
+
+Below are mockups of the package sources page with the added pin feature.
+
+![Options 2](../../meta/resources/PackageSourceMapping/VSOptions1.png)
+
+### Technical Explanation
+
+#### Package Source Mappings Page
+
+When the user opens the package source mapping page in the options dialog, it will automatically display all previous package source mapping configurations. 
+
+When the user adds a package source mapping to their project, the NuGet config will be updated when the user clicks the okay button. The specified package ID will be mapped to the selected source(s) in the config. 

```

Comment:
worth reviiewing the full doc, but we're usually referring to package patterns here.